src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -92,11 +102,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/user-page.html?user=" + recipient);
   }
 
-  /** 
+  /**
    * Private method to recognize the image url in the text message and convert to <img> tag.
    * Message with no img url will not be modified.
    */
   private String extractImgUrl(String text) {
     return (text.replaceAll(this.imgUrlRegex, this.imgUrlReplacement));
   }
+
+  /**
+   * Translates messages based on a URL query parameter.
+   */
+  private void translateMessages(List<Message> messages, String targetLanguageCode) {
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
```

Comment:
Does it handle the case where the source and target languages are the same?

Reply:
Yes! It should. I'll test that.